### PR TITLE
Port dtslint tests to jest, re-root dts-critic tests

### PR DESCRIPTION
**Pull request by sandersn**
dts-critic tests need to be run from DefinitelyTyped-tools/ root now, which also changes a couple of the assertions.

### CODE REVIEW

**Comment by sandersn** [comment]
tslint requires the rules to be built before running tests, but ts-jest doesn't build files first =(

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "pretest": "yarn build",
     "lint": "eslint . --ext .ts",
     "format": "prettier --write 'packages/**/*.ts'",
-    "test": "jest",
+    "test": "tsc -b packages/dtslint && jest",
```
Comment (andrewbranch) [inline package.json:12]: This should already be covered by `"pretest"` 🤔

Comment (sandersn) [reply package.json:12]: Hmm, makes me think that dtslint isn't correctly integrated into the project build.

Comment (sandersn) [reply package.json]: Yep, that was it.

**Pull request merged**